apps/system/js/audio_channel_controller.js

```diff
@@ -9,11 +9,35 @@
   /**
    * AudioChannelController controls the audio channel to
    * play or pause.
+   *
+   * There are two ways to create an AudioChannelController instance.
+   * One is for System app, and the other one is for other apps. 
+   *
+   * Doing `new AudioChannelController('notification')`
+   * to create a controller for System app's notification audio channel.
+   *
+   * Doing `new AudioChannelController(musicApp, contentChannel)`
+   * to create a controller for Music app's content audio channel.
+   * 
+   * @param {AppWindow|String} app
+   * If you want to create a controller for System app's audio channel,
+   * use a String.
+   * If you want to create a controller for other app's audio channel,
+   * use a AppWindow.
+   * @param {BrowserElementAudioChannel} channel
+   * An audio channel belongs to the app.
    */
   var AudioChannelController = function(app, channel) {
     this.app = app;
     this.name = channel.name;
     this._channel = channel;
+    this._isBelongedToSystemApp = app.instanceID === 'systemAppID';
+    if (this._isBelongedToSystemApp) {
+      window.addEventListener('mozChromeEvent', this);
+    } else {
+      app.element.addEventListener('_destroyed', this);
```

Comment:
Out of the scope, but I do remember that I had commented on this for not using the destroyed event.